test.py

```diff
@@ -1654,18 +1309,63 @@ async def find_tests(options: argparse.Namespace) -> None:
                 suite = TestSuite.opt_create(f, options, mode)
                 await suite.add_test_list()
 
-    if not TestSuite.test_count():
-        if len(options.name):
-            print("Test {} not found".format(palette.path(options.name[0])))
-            sys.exit(1)
-        else:
-            print(palette.warn("No tests found. Please enable tests in ./configure.py first."))
-            sys.exit(0)
-
     logging.info("Found %d tests, repeat count is %d, starting %d concurrent jobs",
                  TestSuite.test_count(), options.repeat, options.jobs)
     print("Found {} tests.".format(TestSuite.test_count()))
 
+def run_pytest(options: argparse.Namespace, run_id: int):
+    tmp_dir = pathlib.Path(options.tmpdir)
+    allure_dir = tmp_dir / 'allure'
+    expression = ''
+    if options.name:
+        expression += ' '.join(options.name)
+    expression += f'{"and not ".join(options.skip_patterns)}' if options.skip_patterns else ''
+    modes = ' '.join([f'--mode={mode}' for mode in options.modes])
+    args = [
+        'python',
+        '-m',
+        'pytest',
+        "-s",  # don't capture print() output inside pytest
+        '--color=yes',
+        "--log-level=DEBUG",  # Capture logs
+        '-v',
+        "--junit-xml={}".format(tmp_dir / f'pytest_cpp_{run_id}.xml'),
+        "-rs",
+        "--run_id={}".format(run_id),
+        f'--maxfail={options.max_failures}',
+        modes,
+        f'-n{int(options.jobs)}',
+    ]
+    if options.list_tests:
+        args.extend(['--collect-only', '-q'])
+    if options.gather_metrics:
+        args.append('--gather-metrics')
+    if len(expression) > 1:
+        args.extend(['-k',expression])
+    args.append(f"--alluredir={allure_dir}")
+    if not options.save_log_on_success:
+        args.append("--allure-no-capture")
+    if options.markers:
+        args.append(f"-m={options.markers}")
+    args.append('test/boost test/unit test/ldap test/raft')
```

Comment:
should be constant  PYTEST_FOLDERS  = 'test/boost test/unit test/ldap test/raft'

also not sure regarding the Dummy solution. For me,  better just have a hardcoded TEST_PY_FOLDERS list to save test collect time 

@enaydanov WDYT?

Reply:
As discussed, I'll rename the suite.yaml to test_config.yaml and get rid of the Dummy class.